### PR TITLE
Detect deltaMode - fixes IE11/FF49/Chrome54 on Win7, see #9526

### PR DESCRIPTION
**Pull request by fredludlow**
See #9526

### CODE REVIEW

File: examples/js/controls/TrackballControls.js [hunk 1]
```diff
@@ -473,7 +473,7 @@ THREE.TrackballControls = function ( object, domElement ) {
 		event.preventDefault();
 		event.stopPropagation();
 
-		_zoomStart.y -= event.deltaY * 0.01;
+		_zoomStart.y -= event.deltaY * ( ( event.deltaMode )  ? 0.01 : 0.0001 );
```
Comment (mrdoob) [inline examples/js/controls/TrackballControls.js]: One could think that `event.deltaMode` is a boolean. How about...

```js
( event.deltaMode === 0 ? 0.0001 : 0.01 )
```

**Comment by fredludlow** [comment]
Or even a  switch  case?  (value of 2 means unit is pages). Let me know which of those you prefer and I'll update

**Comment by mrdoob** [comment]
Hmm... In which case would `event.deltaMode` be `2`?

**Comment by mrdoob** [comment]
Yeah... A switch would be the safest indeed. So we can ignore `2`.

**Comment by fredludlow** [comment]
deltaMode takes 0 for pixels, 1 for lines and 2 for pages
https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaMode

Without knowing of a platform that produces a value of 2 it's hard to test but we could assume a page is about 30 lines (approx order of magnitude) and scale accordingly?

**Comment by mrdoob** [comment]
I think I would ignore `2` for now.

**Comment by fredludlow** [comment]
Okay, I've updated it with a switch, can handle case `2` if anyone ever finds a platform that generates it! 

Also changed the scaling for case `0` to 0.00025 so that FF, Chrome and IE feel more consistent.

**Comment by MrOpposite** [comment]
I've successfully managed to get a WheelEvent with deltaMode: 0 on chrome in windows 10.

This is the result of my tests (i simply did `window.addEventListener("wheel",(e)=>console.log(e))` in this browser-window)
image

This is the setting I changed in windows. Notice the "Roll the mousewheel to scroll" being set to "One screen at a time"
image

I'm unsure what needs testing on this issue though, so I couldn't provide anything in that context.

**Comment by fredludlow** [comment]
Thanks @MrOpposite. So... if `deltaMode === 2` then `deltaY` is approximately 1/-1. For `deltaMode=== 0` `deltaY ~= 100`. So I'll update with a `case 2` and scale by 0.025..?

**Comment by MrOpposite** [comment]
Are we sure we want to use the deltaY as a scalefactor for zoom at all? Since the deltaY value is defined by how much a user wants a page to scroll up/down, rather than how much to zoom in an application like this. 

I'd actually advocate ignoring the deltaMode entirely and just use `Math.sign(deltaY)` in order to get the direction and dictate the sensitivity ourselves entirely without relying on sane settings of the client.

**Comment by mrdoob** [comment]
> I'd actually advocate ignoring the deltaMode entirely and just use `Math.sign(deltaY)` in order to get the direction and dictate the sensitivity ourselves entirely without relying on sane settings of the clients OS.

We lose smooth scrolling (smooth zooming) if we do that...

**Comment by MrOpposite** [comment]
Ah, yeah, that's a very good reason to keep battle this issue while still using the value of deltaY...

**Comment by fredludlow** [comment]
I've updated my branch to what I suggest we do for now (handle all three modes and scale accordingly), at least until we get another report of a system this doesn't work on. Very open to suggestions/instructions to do it another way if preferred, @mrdoob?

**Comment by mrdoob** [comment]
LGTM!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**